### DISCUSSION TITLE
Current state of XMI export feature?

### DISCUSSION THREAD

**Discussion by ebousse** [gaphor/gaphor#2026] [Q&A]
I'm playing around with the XMI export feature of Gaphor. What is the current state of this feature? With my few attempts I could  only export UML Class Diagrams, not Use Cases for instance. 

Moreover, I was surprised to see the following header in the generated XMI:
```xml
<XMI xmi.version="2.1" xmlns:xmi="http://schema.omg.org/spec/XMI/2.1" xmlns:UML="http://schema.omg.org/spec/UML/2.1">
```

I don't reckon that `http://schema.omg.org/spec/XMI/2.1` and `http://schema.omg.org/spec/UML/2.1` are the correct namespaces URI for UML? At least in the official [UML 2.5.1 Abstract Syntax Metamodel](https://www.omg.org/spec/UML/20161101/UML.xmi), the URI for the UML namespace is `http://www.omg.org/spec/UML/20161101`, but maybe I'm missing something.

Lastly, I was unable to import an exported XMI into [Eclipse Modeling Tools](https://www.eclipse.org/downloads/packages/release/2022-12/r/eclipse-modeling-tools). My attempt was to simply rename the extension `.xmi` into `.uml` as required by this IDE to be able to open a UML XMI model, but I end up with a very weird glitched model containing only `AnyType` elements, like this:

![image](https://user-images.githubusercontent.com/5868014/213865875-f5d617e9-1da4-41f9-9955-df45a97aa588.png)

Is there a list of tools compatible with the XMI files that Gaphor can export?

**Comment by danyeaw** [comment]
Hi @ebousse, I think you have found some limitations of our XMI export, thanks for that. I would guess that this isn't a feature that is tested very often and could use some help with updates. Would you mind please submitting an issue or two that tracks updates to this? We would also be glad to have some help improving it. :+1:

**Reply by RichardB01** [reply]
Hi Dany, thank you so much for gaphor it is epic and i really enjoy using it. I've been playing with the export XMI feature on OS X but i don't think it actually works, is it supposed to?

**Reply by danyeaw** [reply]
Hey @Bambofy, yes, the intention is that it works. I haven't tested it too much myself, but we are glad to have bug reports with what you find

**Reply by bgkerion** [reply]
Hi @danyeaw, I looked for the XMI export feature in gaphor 3.1.0 with no luck. Has it been removed?

**Reply by danyeaw** [reply]
Yes, it was removed with https://github.com/gaphor/gaphor/pull/3444
